src/dist/replication/meta_server/meta_bulk_load_service.cpp

```diff
@@ -805,6 +805,46 @@ void bulk_load_service::on_partition_ingestion_reply(error_code err,
                                                      const gpid &pid)
 {
     // TODO(heyuchen): TBD
+    // if meet 2pc error, ingesting will rollback to downloading, no need to retry here
+    if (err != ERR_OK) {
+        derror_f("app({}) partition({}) ingestion files failed, error = {}", app_name, pid, err);
+        tasking::enqueue(
+            LPC_META_STATE_NORMAL,
+            _meta_svc->tracker(),
+            std::bind(&bulk_load_service::try_rollback_to_downloading, this, app_name, pid));
+        return;
+    }
+
+    if (resp.err == ERR_TRY_AGAIN && resp.rocksdb_error != 0) {
+        derror_f("app({}) partition({}) ingestion files failed while empty write, rocksdb error = "
+                 "{}, retry it later",
+                 app_name,
+                 pid,
+                 resp.rocksdb_error);
+        tasking::enqueue(LPC_BULK_LOAD_INGESTION,
+                         _meta_svc->tracker(),
+                         std::bind(&bulk_load_service::partition_ingestion, this, app_name, pid),
+                         0,
+                         std::chrono::milliseconds(10));
```

Comment:
What 10ms supposed to do? Will it make any sense while rocksdb failing to ingest files?

Reply:
This condition will be happened when empty write failed before ingestion, we should retry this request. The delay time I set 10ms to let retry quickly, because current replica has already rejected normal write requests.

Reply:
So why not retry immediately?